### PR TITLE
100 poc scheduled report configurations

### LINKED ISSUE #100: POC - Scheduled Report Configurations
## Is your feature request related to a problem? Please describe

Create a mock up for scheduled report config.

### PR DESCRIPTION
**Pull request by jmbrunskill**
Closes #100 

# 👩🏻‍💻 What does this PR do?
 

Configuration UI mock up for managing scheduled reports

Kapture 2023-09-04 at 09 40 56

# 🧪 How has/should this change been tested?

This is mostly ui at the moment, some mocked data

### CODE REVIEW

File: backend/graphql/types/src/types/notification_config.rs [hunk 1]
```diff
@@ -96,18 +96,21 @@ impl NotificationConfigConnector {
 #[serde(rename_all = "SCREAMING_SNAKE_CASE")]
 pub enum ConfigKind {
     ColdChain,
+    Scheduled,
```
Comment (jmbrunskill) [inline backend/graphql/types/src/types/notification_config.rs:99]: I wonder if we should just make this a string, as plugins might have their own configuration types? Thoughts?

File: frontend/packages/system/src/Notifications/Modals/Scheduled/parseConfig.ts [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+export function buildScheduledNotificationInputs(
+  config: ScheduledNotification
+): {
+  create: CreateNotificationConfigInput;
+  update: UpdateNotificationConfigInput;
+} {
+  const input = {
+    id: config.id,
+    title: config.title,
+    configurationData: JSON.stringify(config),
```
Comment (jmbrunskill) [inline frontend/packages/system/src/Notifications/Modals/Scheduled/parseConfig.ts:62]: This means that all the config gets saved as JSON, including `id` for example.
I think this is fine/good?
Previous destructuring approach was easy to miss and then loose new params...

File: frontend/packages/system/src/Notifications/components/SqlQuerySelector.tsx [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+interface SqlQuery {
+  id: string;
+  name: string;
+  query: string;
+  parameters: string[];
+}
+
+type QueryListProps = { records: SqlQuery[] };
+
+export const SqlQuerySelector: FC<QueryListProps> = ({ records }) => {
```
Comment (jmbrunskill) [inline frontend/packages/system/src/Notifications/components/SqlQuerySelector.tsx:20]: Right now this just has a dummy list, but assume we'll upgrade this component to have selection from the sql list (when that's created)

File: frontend/packages/system/src/Notifications/types.ts [hunk 4]
```diff
@@ -10,7 +10,7 @@ export interface BaseNotificationConfig extends BaseConfig {
 }
 
 export interface CCNotification extends BaseNotificationConfig {
-  kind: ConfigKind.ColdChain;
```
Comment (jmbrunskill) [inline frontend/packages/system/src/Notifications/types.ts:13]: I think this was supposed to be discriminating field or some such typescript thing.
Have fallen back to the just the `ConfigKind` for now

**Review by jmbrunskill — COMMENTED** [review]
I think this is ready for review.
Obviously a lot of unfinished functionality but probably better to create specific issues for things we need to add and work on them piece by piece, and allow us to merger open branches.

File: frontend/packages/common/src/types/schema.ts [hunk 5]
```diff
@@ -1,18 +1,31 @@
 export type Maybe<T> = T | null;
 export type InputMaybe<T> = Maybe<T>;
```
Comment (andreievg) [inline frontend/packages/common/src/types/schema.ts:2]: did you upgrade codegen, or manually save this ?

File: frontend/packages/system/src/Notifications/Modals/Scheduled/parseConfig.ts (hunk 2 above)
Comment (andreievg) [reply frontend/packages/system/src/Notifications/Modals/Scheduled/parseConfig.ts:62]: Yeah I think it's fine

**Review by andreievg — APPROVED** [review]
Look good thanks, one change request, but that can be done in a separat PR, and that's casting null to date.

I also have [suggestion](https://github.com/openmsupply/notify/compare/100-poc-scheduled-report-configurations...100-poc-scheduled-report-configurations(Simplify-notification-config-parsing)), see comments in [commit](https://github.com/openmsupply/notify/commit/97697e7505a052ae3d8ecc6adfbdf3abd5fb5abf)

**Review by andreievg — APPROVED** [review]
Look good thanks, one change request, but that can be done in a separat PR, and that's casting null to date.

I also have [suggestion](https://github.com/openmsupply/notify/compare/100-poc-scheduled-report-configurations...100-poc-scheduled-report-configurations(Simplify-notification-config-parsing)), see comments in [commit](https://github.com/openmsupply/notify/commit/97697e7505a052ae3d8ecc6adfbdf3abd5fb5abf)

**Review by andreievg — APPROVED** [review]
Look good thanks, one change request, but that can be done in a separat PR, and that's casting null to date.

I also have [suggestion](https://github.com/openmsupply/notify/compare/100-poc-scheduled-report-configurations...100-poc-scheduled-report-configurations(Simplify-notification-config-parsing)), see comments in [commit](https://github.com/openmsupply/notify/commit/97697e7505a052ae3d8ecc6adfbdf3abd5fb5abf)

File: backend/graphql/types/src/types/notification_config.rs (hunk 1 above)
Comment (andreievg) [reply backend/graphql/types/src/types/notification_config.rs]: hmm, i think for now it's ok to be static with enums. If front end configurations for those configs are done with front end plugins, then each a string would be the way to go

File: frontend/packages/common/src/types/schema.ts (hunk 5 above)
Comment (jmbrunskill) [reply frontend/packages/common/src/types/schema.ts]: Must have saved it!

File: frontend/packages/system/src/Notifications/Modals/Scheduled/ScheduledNotificationEditForm.tsx [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+      <Typography
+        sx={{ fontWeight: 700, fontSize: '13px', marginBottom: '2px' }}
+      >
+        Schedule
+      </Typography>
+      <Typography sx={{ fontSize: '10px' }}>Starting From</Typography>
+      <DateTimeInput
+        onChange={d =>
+          onUpdate({
+            scheduleStartTime: d as ScheduledNotification['scheduleStartTime'],
```
[the remark this answers is not in the document]
Comment (jmbrunskill) [reply frontend/packages/system/src/Notifications/Modals/Scheduled/ScheduledNotificationEditForm.tsx:107]: Issue raised #135

**Pull request merged**